src/lib/snarky_blake2/uint32.ml

```diff
@@ -0,0 +1,158 @@
+open Core_kernel
```

Comment:
Yeah I guess I could have added an add function to that -- I don't have a strong reason. I just took the structure of zcash's bellman implementation and liked the idea of using an array instead of a list. I would actually vote long term for keeping this as a paradigm and renovating the bits module as it's gotten pretty crufty.